checkpoint_test.go

```diff
@@ -200,3 +206,105 @@ func TestCheckpointRestoreNewContainer(t *testing.T) {
 	}
 	<-statusC
 }
+
+func TestCheckpointLeaveRunning(t *testing.T) {
+	if testing.Short() {
+		t.Skip()
+	}
+	if !supportsCriu {
+		t.Skip("system does not have criu installed")
+	}
+	client, err := New(address)
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer client.Close()
+
+	var (
+		ctx, cancel = testContext()
+		id          = t.Name()
+	)
+	defer cancel()
+
+	image, err := client.GetImage(ctx, testImage)
+	if err != nil {
+		t.Error(err)
+		return
+	}
+	spec, err := GenerateSpec(WithImageConfig(ctx, image), WithProcessArgs("sleep", "100"))
+	if err != nil {
+		t.Error(err)
+		return
+	}
+	container, err := client.NewContainer(ctx, id, WithSpec(spec), WithNewRootFS(id, image))
+	if err != nil {
+		t.Error(err)
+		return
+	}
+	defer container.Delete(ctx, WithRootFSDeletion)
+
+	task, err := container.NewTask(ctx, empty())
+	if err != nil {
+		t.Error(err)
+		return
+	}
+	defer task.Delete(ctx)
+
+	statusC := make(chan uint32, 1)
+	go func() {
+		status, err := task.Wait(ctx)
+		if err != nil {
+			t.Error(err)
+		}
+		statusC <- status
+	}()
+
+	if err := task.Start(ctx); err != nil {
+		t.Error(err)
+		return
+	}
+
+	if _, err := task.Checkpoint(ctx); err != nil {
+		t.Error(err)
+		return
+	}
+
+	status, err := task.Status(ctx)
+	if err != nil {
+		t.Error(err)
+		return
+	}
+	if status != Running {
+		t.Errorf("expected status %q but received %q", Running, status)
+		return
+	}
+
+	if err := task.Kill(ctx, syscall.SIGKILL); err != nil {
+		t.Error(err)
+		return
+	}
+
+	<-statusC
+}
+
+func WithExit(r *tasks.CheckpointTaskRequest) error {
```

Comment:
I'm not sure I understand: https://github.com/mlaventure/containerd/commit/44dfa4e060f4a06f09268dedbf3360b2657f7328